cmd/api/database.go

```diff
@@ -125,22 +106,44 @@ func listBooks(name string, minPrice32, maxPrice32 float32, sortBy, sortDirectio
 }
 
 /* Stores the book into the database, checks and returns it if succeed. */
-func storeOnDB(newBook Book) (Book, error) {
+func storeOnDB(bookEntry Book) (Book, error) {
 	sqlStatement := `
 	INSERT INTO bookstable (id, name, price, inventory, created_at, updated_at)
 	VALUES ($1, $2, $3, $4, $5, $6)
 	RETURNING *`
-	createdRow := dbObjectGlobal.QueryRow(sqlStatement, newBook.ID, newBook.Name, *newBook.Price, *newBook.Inventory, newBook.CreatedAt, newBook.UpdatedAt)
+	createdRow := dbObjectGlobal.QueryRow(sqlStatement, bookEntry.ID, bookEntry.Name, *bookEntry.Price, *bookEntry.Inventory, bookEntry.CreatedAt, bookEntry.UpdatedAt)
 	var bookToReturn Book
 	err := createdRow.Scan(&bookToReturn.ID, &bookToReturn.Name, &bookToReturn.Price, &bookToReturn.Inventory, &bookToReturn.CreatedAt, &bookToReturn.UpdatedAt)
 	if err != nil {
 		switch err {
 		case sql.ErrNoRows:
-			return Book{}, fmt.Errorf("storing on db: %w", errBookNotFound)
+			return Book{}, fmt.Errorf("storing on db: %w", errResponseBookNotFound)
```

Comment:
You're right. The book the was supposed to be created was not found. In other words: was not created. Could be errBookNotCreated. Buuut, we named it as errBookNotFound because it is an ErrNoRows, and we have other case to unexpected errors when the book is not created either. Furthermore, we are already returning a 500 for both cases.
I see that the big mistake here is to replace errBookNotFound (an internal thing) by an errResponse, that is aiming at a http RESPONSE. Could we just go back to errBookNotFound?

Reply:
Just wrap the error and return it. Our handler will take care of making it a 500.

Reply:
What should I maintain: errBookNotFound or errResponseBookNotFound?

Reply:
Or neither one? We could take out the switch and just return a wrapped err.

Reply:
Yep, there is no need to check for a specific error. Any error will cause a 500.